frontends/p4/sideEffects.cpp

```diff
@@ -160,7 +160,9 @@ const IR::Node* DoSimplifyExpressions::preorder(IR::StructExpression* expression
         auto path = addAssignment(expression->srcInfo, tmp, v->expression);
         typeMap->setType(path, t);
         v = new IR::NamedExpression(v->name, path);
+        vec.push_back(v);
     }
+    expression->components = vec;
```

Comment:
This seems pointless?  `v` is already a reference to the pointer in expression->components.  Simpler would be to just get rid of the unused `vec`

Reply:
Actually this does not work because of issue https://github.com/p4lang/p4c/issues/43